.vitepress/components/SidebarItemOverride.vue

```diff
@@ -0,0 +1,313 @@
+<script setup lang="ts">
+/**
+ * ok so because this is recursive I can't simply call the inner component
+ * but rather have to remake the entire component with the added slot
+ * ! if the vitepress version ever gets updated this won't update along with it
+ */
+
+import { computed } from "vue";
+import type { DefaultTheme } from "vitepress/theme";
+import { useSidebarControl } from "vitepress/dist/client/theme-default/composables/sidebar";
+import VPLink from "vitepress/dist/client/theme-default/components/VPLink.vue";
+import VPBadge from "vitepress/dist/client/theme-default/components/VPBadge.vue";
+
+/**
+ * Check if a DD/MM/YYYY date is older than one month
+ * @author Juknum
+ * @param date DD/MM/YYYY date to check
+ * @returns whether it is
+ */
+function isNew(date?: string) {
+	if (date === undefined) return false;
+
+	const delayedDate = new Date();
+	delayedDate.setMonth(delayedDate.getMonth() - 1);
+	const postDate = new Date();
+	const [day, month, year] = date.split("/").map((el) => Number.parseInt(el, 10));
+	postDate.setDate(day);
+	postDate.setMonth(month - 1);
+	postDate.setFullYear(year);
+
+	if (isNaN(postDate.getTime())) return false;
+
+	/**
+	 * If the post date is greater than today's date - 1 month, show the span
+	 */
+	return postDate.getTime() > delayedDate.getTime();
+}
+
+interface SidebarItemOverride extends DefaultTheme.SidebarItem {
+	date?: string;
+	archived?: boolean;
+	deprecated?: boolean;
+	items?: SidebarItemOverride[];
+}
+
+const props = defineProps<{
+	item: SidebarItemOverride;
+	depth: number;
+}>();
+
+const { collapsed, collapsible, isLink, isActiveLink, hasActiveLink, hasChildren, toggle } =
+	useSidebarControl(computed(() => props.item));
+
+const sectionTag = computed(() => (hasChildren.value ? "section" : `div`));
+
+const linkTag = computed(() => (isLink.value ? "a" : "div"));
+
+const textTag = computed(() => {
+	return !hasChildren.value ? "p" : props.depth + 2 === 7 ? "p" : `h${props.depth + 2}`;
+});
+
+const itemRole = computed(() => (isLink.value ? undefined : "button"));
+
+const classes = computed(() => [
+	[`level-${props.depth}`],
+	{ collapsible: collapsible.value },
+	{ collapsed: collapsed.value },
+	{ "is-link": isLink.value },
+	{ "is-active": isActiveLink.value },
+	{ "has-active": hasActiveLink.value },
+]);
+
+function onItemInteraction(e: MouseEvent | Event) {
+	if ("key" in e && e.key !== "Enter") {
+		return;
+	}
+	!props.item.link && toggle();
+}
+
+function onCaretClick() {
+	props.item.link && toggle();
```

Comment:
don't try to be as low code as possible, minifier will do the job for you, increase readability instead
```suggestion
	// if sublist and (click or "Enter" key)
	if (!props.item.link && (!("key" in e) || e.key === "Enter"))
		toggle();
}

function onCaretClick() {
	if(props.item.link)
		toggle();
```